lib/google_contacts_api/api.rb

```diff
@@ -29,26 +29,101 @@ def get(link, params = {}, headers = {})
         raise UnauthorizedError if defined?(e.response) && self.class.parse_response_code(e.response) == 401
         raise e
       end
-      
+
       # OAuth 1.0 uses Net::HTTP internally
       raise UnauthorizedError if result.is_a?(Net::HTTPUnauthorized)
       result
     end
 
+    # these two methods will replace my previous two, but they aren't wired up to
+    # the rest of the code yet
     # Post request to specified link, with query params
-    # Not tried yet, might be issues with params
-    def post(link, params = {}, headers = {})
-      raise NotImplementedError
-      params["alt"] = "json"
-      @oauth.post("#{BASE_URL}#{link}?#{params.to_query}", headers)
+    def post_v2(link, params = {}, headers = {})
+      merged_params = params_with_defaults(params)
+      uri = "#{BASE_URL}#{link}?#{merged_params.to_query}"
+      begin
+        response = @oauth.post(uri, headers)
+      rescue => e
+        raise UnauthorizedError if defined?(e.response) && self.class.parse_response_code(e.response) == 401
```

Comment:
It more than likely means he doesn't test for the case where `defined?(e.response)` returns false. I'd be very wary of getting rid of that check. I just think it should be

    if e.response && (self.class.parse_response_code(e.response) == 401)
      ...